frontend/src/model/beta/accessRequests/AccessRequestFormEditPage.tsx

```diff
@@ -0,0 +1,127 @@
+import { Box, Button, Divider, Stack, Typography } from '@mui/material'
+import { useTheme } from '@mui/material/styles'
+import { patchAccessRequest, useGetAccessRequest } from 'actions/accessRequest'
+import { useContext, useEffect, useState } from 'react'
+import UnsavedChangesContext from 'src/contexts/unsavedChangesContext'
+import { getErrorMessage } from 'utils/fetcher'
+
+import { useGetSchema } from '../../../../actions/schema'
+import { useGetUiConfig } from '../../../../actions/uiConfig'
+import { AccessRequestInterface, SplitSchemaNoRender } from '../../../../types/interfaces'
+import { getStepsData, getStepsFromSchema } from '../../../../utils/beta/formUtils'
+import Loading from '../../../common/Loading'
+import ModelCardForm from '../../../Form/beta/ModelCardForm'
+import MessageAlert from '../../../MessageAlert'
+
+type FormEditPageProps = {
+  accessRequest: AccessRequestInterface
+}
+
+export default function AccessRequestFormEditPage({ accessRequest }: FormEditPageProps) {
+  const [isEdit, setIsEdit] = useState(false)
+  const [splitSchema, setSplitSchema] = useState<SplitSchemaNoRender>({ reference: '', steps: [] })
+  const [errorText, setErrorText] = useState('')
+
+  const { schema, isSchemaLoading, isSchemaError } = useGetSchema(accessRequest.schemaId)
+  const { mutateAccessRequest } = useGetAccessRequest(accessRequest.modelId, accessRequest.id)
+  const { uiConfig: _uiConfig, isUiConfigLoading, isUiConfigError } = useGetUiConfig()
+
+  const { setUnsavedChanges } = useContext(UnsavedChangesContext)
+  const theme = useTheme()
+
+  async function onSubmit() {
+    if (schema) {
+      const data = getStepsData(splitSchema, true)
+      const res = await patchAccessRequest(accessRequest.modelId, accessRequest.id, data)
+      if (!res.ok) {
+        setErrorText(await getErrorMessage(res))
+      } else {
+        setIsEdit(false)
+        setErrorText('')
+        mutateAccessRequest()
+      }
+    }
+  }
+
+  function onCancel() {
+    if (schema) {
+      mutateAccessRequest()
+      const steps = getStepsFromSchema(schema, {}, ['properties.contacts'], accessRequest.metadata)
```

Comment:
We already do run this exact logic to update the SplitSchema in the `useEffect` below. I think it can be removed from here because `mutateAccessRequest()` will trigger that effect to run if needed.

Reply:
The `mutateAccessRequest` isn't being called onCancel if we leave this code out

Reply:
I meant keep the mutate here and remove the steps/split schema logic. Can it be trimmed down to this?
```
function handleCancel () {
 ()
  setIsEdit(false)
}
```

That way we call mutate and set the component back to readonly mode and the `useEffect` on line 58 should handle the steps/split schema logic due to `accessRequest` being in it's dependency array (the `accessRequest` will be "changed" due to the mutate call, triggering the effect)

Reply:
I meant keep the mutate here and remove the steps/split schema logic. Can it be trimmed down to this?
```
function handleCancel () {
 mutateAccessRequest()
  setIsEdit(false)
}
```

That way we call mutate and set the component back to readonly mode and the `useEffect` on line 58 should handle the steps/split schema logic due to `accessRequest` being in it's dependency array (the `accessRequest` will be "changed" due to the mutate call, triggering the effect)

Reply:
I meant keep the mutate here and remove the steps/split schema logic. Can it be trimmed down to this?
```
function handleCancel () {
  mutateAccessRequest()
  setIsEdit(false)
}
```

That way we call mutate and set the component back to readonly mode and the `useEffect` on line 58 should handle the steps/split schema logic due to `accessRequest` being in it's dependency array (the `accessRequest` will be "changed" due to the mutate call, triggering the effect)

Reply:
Also, if this logic is taken from elsewhere, can we make the same change there too. This split schema logic is notoriously confusing and if we can simplify it a bit I think we should.

Reply:
Due to the accessRequest not being changed while working on the form, the mutate call wasn't having any effect. The following should work as we want:
```
  const resetForm = useCallback(() => {
    if (schema) {
      const steps = getStepsFromSchema(schema, {}, ['properties.contacts'], accessRequest.metadata)

      for (const step of steps) {
        step.steps = steps
      }

      setSplitSchema({ reference: schema.id, steps })
    }
  }, [accessRequest.metadata, schema])

  function handleCancel() {
    setIsEdit(false)
    resetForm()
  }

  useEffect(() => {
    resetForm()
  }, [resetForm])
```